### DISCUSSION TITLE
ArrayTable 操作栏，实现自定义内容。[https://antd.formilyjs.org/zh-CN/components/array-table]

### DISCUSSION THREAD

**Discussion by kongjunAI** [alibaba/formily#3703] [Q&A 2.x] [answered]
ArrayTable 操作栏实现自定义内容。比如：删除操作，需要自定义文本内容，并且删除前加上确定删除提示框。

![image](https://user-images.githubusercontent.com/76086371/217414727-41e3929a-9148-47a2-a4bb-ffb360229f47.png)

官网的实现：https://codesandbox.io/s/xdfdrh

```react
 <SchemaField.Void
    x-component="ArrayTable.Column"
    x-component-props={{
      title: 'Operations',
      dataIndex: 'operations',
      width: 200,
      fixed: 'right',
    }}
  >
    <SchemaField.Void x-component="FormItem">
      <SchemaField.Void x-component="ArrayTable.Remove" />
    </SchemaField.Void>
  </SchemaField.Void>
```

**Comment by xiaosi-ss** [comment]
如何自定义删除组件？

**Reply by kongjunAI** [reply]
是的，是的。

**Reply by kongjunAI** [reply]
默认是这个组件：<SchemaField.Void x-component="ArrayTable.Remove" />

**Reply by kongjunAI** [reply]
有解法嘛？🙏

**Comment by xiaosi-ss** [comment]
这不是很简单  你把ArrayTable.Remove换成你自己写component的不就行了

**Reply by kongjunAI** [reply]
是的，这是一种方法。但是需要我手动修改数据源才能移除某一行数据，formily 并没有提供类似与 remove 的方法。在 antd Form.List 中就有类似的方法。
![image](https://user-images.githubusercontent.com/76086371/217462028-254be41b-a1f5-4d90-a61e-cc811bede60f.png)

**Comment by xiaosi-ss** [comment, marked as answer]
有的啊……
![image](https://user-images.githubusercontent.com/7309546/217475498-46977be0-be42-4542-936d-8dca30e4a3e3.png)

**Reply by kongjunAI** [reply]
已解决，感谢 🙏。

**Reply by CodeAndMoney** [reply]
可以说一下怎么解决的吗，感谢
